aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -261,24 +287,58 @@ Tensor& baddbmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor&
 
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(!result_->is_conj());
 
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, self.scalar_type(), "baddbmm_cuda", [&] {
-    scalar_t alpha_val = alpha.to<scalar_t>();
-    scalar_t beta_val = beta.to<scalar_t>();
-    scalar_t* batch1_ptr = batch1_->data_ptr<scalar_t>();
-    scalar_t* batch2_ptr = batch2_->data_ptr<scalar_t>();
-    scalar_t* result_ptr = result_->data_ptr<scalar_t>();
-    at::cuda::blas::bgemm<scalar_t>(
-      transpose_batch1 ? batch1_->is_conj() ? 'c' : 't' : 'n',
-      transpose_batch2 ? batch2_->is_conj() ? 'c' : 't' : 'n',
-      m, n, k,
-      alpha_val,
-      batch1_ptr, lda, batch1_->strides()[0],
-      batch2_ptr, ldb, batch2_->strides()[0],
-      beta_val,
-      result_ptr, ldc, result_->strides()[0],
-      num_batches
-    );
-  });
+  if (self.scalar_type() == ScalarType::Long || self.scalar_type() == ScalarType::Int) {
+    AT_DISPATCH_INDEX_TYPES(self.scalar_type(), "baddmm_cuda_int", [&] {
+      using scalar_t = index_t;
+
+      scalar_t alpha_val = alpha.to<scalar_t>();
+      scalar_t beta_val = beta.to<scalar_t>();
+
+      auto batch1_stride = batch1_->strides()[0];
+      auto batch2_stride = batch2_->strides()[0];
+      auto result_stride = result_->strides()[0];
+
+      auto batch1_data = batch1_->data_ptr<scalar_t>();
+      auto batch2_data = batch2_->data_ptr<scalar_t>();
+      auto result_data = result_->data_ptr<scalar_t>();
+
+      for (int64_t batch = 0; batch < num_batches; ++batch) {
+        scalar_t * batch1_working_ptr = &batch1_data[batch * batch1_stride];
+        scalar_t * batch2_working_ptr = &batch2_data[batch * batch2_stride];
+        scalar_t * result_working_ptr = &result_data[batch * result_stride];
+          at::cuda::blas::integer_gemm<scalar_t>(
```

Comment:
This loop will work if the batch dimensions are contiguous. Do you check for each input `x` if `(x.contiguous() or x.t().contiguous())` evaluates to `True`?

Reply:
I've added a check in the code and will add a test in the test files too.